### ISSUE TITLE
[G11] Internal PE-D

### ISSUE THREAD

**Issue by yusufaine** [AY2122S2-CS2103T-T17-2/tp#240]
Simulate PE-D and report any findings here!

Best to follow a format like

```
## [<Docs/Bug/Flaw>-<specifics>-[additional detail]] <Descriptive one-liner of issue>

Severity: <High, Medium, Low, Very Low>

<Details such as steps to reproduce or screenshots>

[Suggestions]
```

<details>
<summary>Click here to view the examples</summary>

## [Docs-UG-Add] Invalid sample command for `add`. 

Severity: Very low

Sample command was missing one mandatory field. 

---

## [Bug-Find] Separator value does not work. 

Severity: Medium

Find command allows searching of values separated by commas, but the input does not seem to be separated. 

Steps to reproduce:
1. `find test, abc`
2. Returns an error message saying `unable to "test, abc"`. (can include screenshot)

Suggestion:
- Check if the user input that is parsed splits by commas, 
- otherwise update the UG and sample commands to indicate how values are separated. hi
</details>

Massive shout out to everyone who took part in this! ❤️

**Comment by yusufaine** [comment]
# [Docs-UG-Add] Invalid sample command for `add`. 

Severity: Very low

Sample command was missing one mandatory field. 

---

# [Feat-Find] Separator value does not work. 

Severity: Medium

Find command allows searching of values separated by commas, but the input does not seem to be separated. 

Steps to reproduce:
1. `find test, abc`
2. Returns an error message saying `unable to "test, abc"`. (can include screenshot)

**Comment by pyk595** [comment]
Test!

**Comment by pyk595** [comment]
[Bug-reset- prompt] No prompts for yes or no
Severity: Low

Features like `clear` or `reset` do not prompt the user when you do not use the `-f` flag
<img width="862" alt="image" src="https://user-images.githubusercontent.com/77255760/162452375-5e47471d-9c24-499e-9ab7-9bcb1333f0f8.png">
<img width="862" alt="image" src="https://user-images.githubusercontent.com/77255760/162452402-5a4b13a0-e808-4cf4-a646-6c28eddcb58d.png">

Suggestion:
Probably just edit the user guide to say that it's a future feature

**Comment by pyk595** [comment]
[Doc-view-recipe does not exist] Filtered list shows wrong message for view error
Severity: Low

From the full list, I saw shoyu ramen, but when I filtered using a keyword like "bacon", and I tried to run `view shoyu ramen` it says the the recipe does not exist, even though it does exist in the recipe book.

Suggestion:
- Explicitly state in the view documentation that it will only display the recipes listed in the side bar currently

**Comment by pyk595** [comment]
[Doc-Command Summary-view] Command summary shows wrong command format
severity: Low

Suggestion: view -x <recipe number from list>

**Comment by pyk595** [comment]
[Bug-Edit-Wrong Message] Edit command shows wrong error message
severity: Low

When I give a wrong prefix in `edit -x`, it'll show an index error message.
<img width="862" alt="image" src="https://user-images.githubusercontent.com/77255760/162459966-8ca2460e-bdba-41bb-b222-df0e95971e06.png">

Suggestions:
Add a check when parsing for an invalid prefix (not sure if you're still using argMultimap, but if you are it'll be a quick fix)

**Comment by pyk595** [comment]
[Bug-view-wrong recipe] Fried rice recipe wrong
severity: Ultra high

Uncle Roger will be disappointed

**Comment by BrendonLau** [comment]
Doc-view-output is different than stated

Severity:  Low

![image](https://user-images.githubusercontent.com/66625773/162484334-4444164a-3bce-4438-afe2-dd8a73fa39a1.png)

Point 3 output is different from point 4

`view -x -1`
![image](https://user-images.githubusercontent.com/66625773/162484463-e4a0ba2d-bd3a-4700-9248-37178079f291.png)

`view -x 100`
![image](https://user-images.githubusercontent.com/66625773/162484664-abd02133-8317-45f0-8a08-33d275010ac0.png)

**Comment by BrendonLau** [comment]
## [Bug-exit-exit the application with extra parameter]

severity: low

`exit 5` will exit the application.

Although it will not affect the user much, it is still a bug since it is not stated in UG?

**Comment by justinyjt** [comment]
### [Mini Bug-cannot remove tag-UG states that tag is optional but app doesn't allow removal of tags via edit command]
Severity: Low

<img width="745" alt="image" src="https://user-images.githubusercontent.com/77252831/162513346-83b0b7a9-0f31-4065-8f71-f60272457a6d.png">

Similar to @pyk595's comment above, the app instead gives the invalid index provided message.

Suggestion: maybe since cant change the code alr can just state in the UG that edit command cannot remove tags and if u wanna do it, need to delete then add etc.###

**Comment by justinyjt** [comment]
[Bug?-Duplicate Content check-duplicate checks only for recipe name]
severity: completely inconsequential

<img width="746" alt="image" src="https://user-images.githubusercontent.com/77252831/162516051-5d05133d-ba7c-4259-8028-8805cb33b1da.png">

I really don't agree with this bug as i don't think anyone would be dumb enough to copy the exact same recipe details and add it in with another name but I realised a lot of people like to use this as a "go-to-bug" cos no one implements it and it's somewhat of a low hanging fruit. So suggestion would be to actively "defend" against this bug by explicitly stating in the UG that only names are checked for duplicates and the content of the recipe is not. So if these guys write this as a bug without checking the ug thoroughly, it will backfire on them. 

TLDR update ug to state that duplicate checks only work for recipe name to set a trap for blur people

p.s. another similar bug is the one where they add additional spaces in the recipe name and call it a bug cos the name looks the same

<img width="747" alt="image" src="https://user-images.githubusercontent.com/77252831/162517207-11091b14-c4fc-4d70-aecb-bbed58318175.png">

command: `edit -x 2 -n Aglio<double space>Olio`

**Comment by petermonky** [comment]
## [Flaw-UI-Text Wrap] Text wrap makes ingredients hard to read

### Severity
Low

### Description
The text for the ingredients wrap around when exceeding the text box, making them somewhat hard to read.

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162556778-a638483a-e5ff-48d0-a2f2-33b160de400d.png">

### Suggestion
Similar to the numbered list in the recipe steps, use a bullet list to list down ingredients.

**Comment by petermonky** [comment]
## [Flaw-UI-Serving size] Serving size box not so obvious

### Severity
Low

### Description
May be a personal thing, but I didn't realise there was a serving size box next to the total time box until I scrolled to the right. Another minor issue that may or may not be relevant to this is that the serving size box gets cut off on the right when the window is not big enough; the latter of the two images below demonstrates this.

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162557048-42c91d0c-9b15-4aca-80f8-5437fbe0bf1d.png">

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162557065-59ded2f4-7a14-4464-b663-0814aa8b6ec8.png">

### Suggestions
* Increase the minimum window width so that the serving size box is clearly displayed upon start up.
* Place the serving size box below the total time box.

**Comment by petermonky** [comment]
## [Flaw?-Default recipes-Vague steps] Vague steps for default list recipes...?

### Severity
Very low

### Description
Steps for some default list recipes seem a bit vague; potentially clusters up application if considered unusable. If only it was this easy to cook 😔

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162557221-6c046f9f-7032-4072-822a-f3a18b98713e.png">

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162557394-670e0eaa-6c22-4bc3-9471-68300099d5f7.png">

### Suggestions
* Perhaps more descriptive steps for default list recipes (e.g. time for each step).

**Comment by petermonky** [comment]
## [Docs-Table of Content] Features could be categorised

### Severity
Low

### Description
The ToC (and I think the UG as a whole) simply lists out all features. I think it could help to categorise the features to whatever scope in which they operate.

<img width="542" alt="image" src="https://user-images.githubusercontent.com/39848410/162557941-f645b0ae-accc-4996-ade0-6d6232d28a64.png">

### Suggestions
* Separate the features to multiple categories (e.g. recipe modifying, general, miscellaneous, etc.).

**Comment by petermonky** [comment]
## [Docs-Table of Content-`exit`] `exit` is capitalised in the ToC

### Severity
Very Low

### Description
The `exit` code text is capitalised.

<img width="456" alt="image" src="https://user-images.githubusercontent.com/39848410/162558053-c2d0adc0-32f0-4a2d-9fb9-4d78dbc3bc20.png">

**Comment by petermonky** [comment]
## [Docs-Quick start] Instructions for Windows users may be fatal

### Severity
High

### Description
As outlined in the UG, the instruction for Windows users is to double-click the file to start the app. This may cause issues on some devices where the app is not given access to system files.

![image](https://user-images.githubusercontent.com/39848410/162558203-98d4abfe-146f-44aa-9190-594a1bb61dcb.png)

This issue does not seem to persist when running the application on Windows with `java -jar McKitchen.jar`.

### Suggestions
* Add extra information in the UG about what to do if the app does not work on Windows (e.g. use `java-jar McKitchen.jar`).

**Comment by petermonky** [comment]
## [Docs-`list` feature] No period at the end of `list` feature description

### Severity
Very low

### Description
The `list` feature description does not appear to have a period at the end like other feature descriptions do.

<img width="762" alt="image" src="https://user-images.githubusercontent.com/39848410/162558609-9f89c2f5-abba-44d9-a0a0-5e10e1693e69.png">

**Comment by petermonky** [comment]
## [Bug?-Recipe time] Invalid recipe time does not show error message

### Severity
Medium

### Description
When editing or adding a recipe with an invalid recipe time (greater than 525600), the error is not displayed.

<img width="750" alt="image" src="https://user-images.githubusercontent.com/39848410/162558775-2b58d253-8a01-4e59-a6e3-601b3d34e9bd.png">

**Comment by kerbysoh** [comment]
[Docs-`exit` feature ] Description is wrong for Mac

Severity: Very Low

<img width="837" alt="Screenshot 2022-04-09 at 10 41 58 PM" src="https://user-images.githubusercontent.com/69448271/162579086-4868b47d-74ce-4de1-8ec7-31acb8ed04d1.png">

Very minor, but the close window button is on the top left for macOS devices.

[Suggestions]

Add more detailed description.

**Comment by kerbysoh** [comment]
[Bug-`add` feature description] Command format description does not work

Severity: Low

<img width="955" alt="Screenshot 2022-04-09 at 10 53 42 PM" src="https://user-images.githubusercontent.com/69448271/162579451-d65bed54-3844-4ce8-a11f-88b3ecb82ecb.png">

The example command does not work, prompting me to add in the amount for each ingredient.

[Suggestions]

Fix example.

**Comment by kerbysoh** [comment]
[Docs-`add` feature description] Add feature example 

Severity: Very Low

<img width="825" alt="Screenshot 2022-04-09 at 10 57 12 PM" src="https://user-images.githubusercontent.com/69448271/162579525-5242bdaf-abf5-43e5-9fcf-d847ec358f5d.png">

Example given does not work, prompting that the receipt already exists. Not a big issue, but would confuse users if they are plainly following guide to familiarise themselves.

[Suggestions]

Use example not in recipe list already

**Comment by yusufaine** [comment]
@justinyjt @petermonky @kerbysoh thanks for catching some of these things and providing suggestions, we really appreciate it guys ❤️ 

We'll be closing this issue now since the submission's tomorrow, good luck to all of us!
